docs/dev/Release.md

```diff
@@ -0,0 +1,365 @@
+        export GPG_TTY=$(tty)
+        ```
+    3. Clean up old release directory and clone a new drill repository
+        ```
+        cd ~/src/release && \
+        rm -rf drill/ && \
+        git clone git@github.com:apache/drill.git && \
+        cd drill
+        ```
+    4. Checkout the current release:
+        ```
+        git checkout <git commit>
+        ```
+    5. Make sure build with tests succeed along with license checks:
+        ```
+        mvn install -Drat.skip=false -Dlicense.skip=false
+        ```
+    6. Enforce memory options:
+        ```
+        MAVEN_OPTS='-Xmx4g -XX:MaxPermSize=512m'
+        ```
+    7. Clear any release history
+        ```
+        mvn release:clean -Papache-release -DpushChanges=false -DskipTests
+        ```
+    8. Delete outgoing release tag if it exists
+        ```
+        git push --delete origin drill-1.17.0
+        git tag -d drill-1.17.0
+        ```
+    9. Do the release preparation
+        ```
+        mvn -X release:prepare -Papache-release -DpushChanges=false -DskipTests -Darguments="-Dgpg.passphrase=${GPG_PASSPHRASE}  -DskipTests=true -Dmaven.javadoc.skip=false" -DreleaseVersion=1.17.0 -DdevelopmentVersion=1.18.0-SNAPSHOT -Dtag=drill-1.17.0
+        ```
+    10. Make sure to change Drill version to the proper one 
+    11. Check that `target` folder contains the following files (with the correct version number):
+        ```
+        apache-drill-1.17.0-src.tar.gz 
+        apache-drill-1.17.0-src.tar.gz.asc
+        apache-drill-1.17.0-src.tar.gz.sha512
+        apache-drill-1.17.0-src.zip
+        apache-drill-1.17.0-src.zip.asc 
+        apache-drill-1.17.0-src.zip.sha512
+        ```
+    12. Verify signature, ensure that GPG key for Apache was used (see details at
+        https://www.apache.org/info/verification.html#CheckingSignatures) and checksum
+        (see details at https://www.apache.org/info/verification.html#CheckingHashes):
+         ```
+        ./tools/release-scripts/checksum.sh target/apache-drill-1.17.0-src.tar.gz
+        ./tools/release-scripts/checksum.sh target/apache-drill-1.17.0-src.zip
+        ./tools/release-scripts/checksum.sh distribution/target/apache-drill-1.17.0.tar.gz
+        ```
+    13. Push the new tag to your personal github repository
+        You should see 2 new commits on top of the branch and a new `drill-1.17.0` tag
+        ```
+        git push origin drill-1.17.0
+        ```
+    14. Perform the release but do the commits to personal repo
+        (Since Apache doesn't allow rollbacks on master if mistakes are made)
+        ```
+        mvn release:perform -DconnectionUrl=scm:git:git@github.com:vvysotskyi/drill.git -DskipTests -Darguments="-Dgpg.passphrase=${GPG_PASSPHRASE} -DskipTests=true -DconnectionUrl=scm:git:git@github.com:vvysotskyi/drill.git"
+        ```
+        If you want additionally check resulting archives and jars, add `-Dmaven.deploy.skip=true` flag to avoid deploying jars to the Nexus repository:
```

Comment:
```suggestion
        If you want to additionally check resulting archives and jars, add `-Dmaven.deploy.skip=true` flag to avoid deploying jars to the Nexus repository:
```